src/dtemplate.d

```diff
@@ -8715,6 +8715,25 @@ public:
         return 0;
     }
 
+    override int apply2(Dsymbol_apply_ft_t fp, void* param)
```

Comment:
OK I checked whole dmd code. Now `Dsymbol.apply` is used only in:

`AggregateDeclaration.determineFields`
`AggregateDeclaration.searchCtor()`

And they are usually invoked after the scope members' `semantic` finished, so today they would already see resolved template mixin members.

Therefore, resolving `TemplateMixin` in its `apply` would have low risk to introduce new forward reference issues.

Let's wait auto-tester result.